zebrad/src/commands/tests.rs

```diff
@@ -0,0 +1,38 @@
+//! Tests for parsing zebrad commands
+
+use clap::Parser;
+
+use crate::commands::ZebradCmd;
+
+use super::EntryPoint;
+
+#[test]
+fn args_with_subcommand_pass_through() {
```

Comment:
It seems like this test is occasionally failing with:
> process didn't exit successfully: `/home/runner/work/zebra/zebra/target/release/deps/zebrad-4d9e827350fddfd8` (signal: 11, SIGSEGV: invalid memory reference)

https://github.com/ZcashFoundation/zebra/actions/runs/5194379481/jobs/9365975834#step:15:3532

Reply:
I think it was because of the `err.exit()` in `process_cli_args()`.

Reply:
It looks like the output of `zebrad help` has been corrupted:

```
Zcash Foundation <zebra@zfnd.org>
brad [OPTIONS] [FILTERS]... [COMMAND]
Commands:
```
https://github.com/ZcashFoundation/zebra/actions/runs/5194379481/jobs/9365975834#step:15:3509

But when I run the test on my x86_64 Linux and M1 macOS I see:
```
Zcash Foundation <zebra@zfnd.org>
Usage: zebrad [OPTIONS] [FILTERS]... [COMMAND]
Commands:
```

I have done thousands of test runs that succeed, with stable Rust 1.70 and nightly Rust 1.72.

Reply:
The previous failure had slightly different memory corruption:

```
Zcash Foundation <zebra@zfnd.org>
sage: zebrad [OPTIONS] [FILTERS]... [COMMAND]
Commands:
```

https://github.com/ZcashFoundation/zebra/actions/runs/5194379481/jobs/9365975834#step:15:3509

Reply:
I don't know if the memory corruption is diagnostic, I think it's also happening in successful test runs:
https://github.com/ZcashFoundation/zebra/actions/runs/5192882552/jobs/9362667847#step:15:3167

Maybe we should add a snapshot test for the usage output to catch bugs like this. (But in another ticket/PR.)

Reply:
> I have done thousands of test runs that succeed, with stable Rust 1.70 and nightly Rust 1.72.

I haven't seen it locally either but I think it happens consistently in CI, the passing tests were skipped.

Reply:
It seems like exiting the test process without warning could cause memory issues and output corruption, so I'm happy to approve this and see if it passes.

If we see any memory errors in the zebrad lib tests after it merges, let's revert this PR.